Source/WebGPU/WGSL/Metal/MetalFunctionWriter.cpp

```diff
@@ -558,6 +542,37 @@ void FunctionDefinitionWriter::visit(const Type* type)
             m_stringBuilder.append(type, "<");
             visit(texture.element);
             m_stringBuilder.append(", access::", mode, ">");
+        },
+        [&](const Reference& reference) {
+            if (reference.accessMode == AccessMode::Read)
+                m_stringBuilder.append("const ");
+            switch (reference.addressSpace) {
+            case AddressSpace::Function:
+                m_stringBuilder.append("thread");
+                break;
+            case AddressSpace::Private:
+                m_stringBuilder.append("private");
+                break;
+            case AddressSpace::Workgroup:
+                m_stringBuilder.append("threadgroup");
+                break;
+            case AddressSpace::Uniform:
+            case AddressSpace::Storage:
```

Comment:
I don't think this is right. I think the right MSL address space for WGSL's `Uniform` is `constant`.

Reply:
Thanks, I wasn't sure about that. I'll update it